src/modules/user/commands/delete-user/delete-user.service.ts

```diff
@@ -1,8 +1,11 @@
 import { UserRepositoryPort } from '@modules/user/database/user.repository.port';
+import { CommandHandler } from '@nestjs/cqrs';
+import { UserRepository } from '../../database/user.repository';
 import { DeleteUserCommand } from './delete-user.command';
 
+@CommandHandler(DeleteUserCommand)
 export class DeleteUserService {
-  constructor(private readonly userRepo: UserRepositoryPort) {}
+  constructor(private readonly userRepo: UserRepository) {}
```

Comment:
Good point.
I made the change, so that the dependency injection would inject an implementation here. With the current code I get this error:
```
TypeError: Cannot read property 'findOneByIdOrThrow' of undefined
    at DeleteUserService.execute (.../domain-driven-hexagon/src/modules/user/commands/delete-user/delete-user.service.ts:10:39)
```

Do you have a suggestion on how to resolve it?

Reply:
You have to inject it properly:
```typescript
  constructor(
      @Inject(UserRepository)
      private readonly userRepo: UserRepositoryPort
) {}
```

Reply:
Perfect. That resolves it.